cms/static/js/views/metadata.js

```diff
@@ -33,12 +33,17 @@ function(BaseView, _, MetadataModel, AbstractEditor, FileUpload, UploadDialog, V
                             'Float': 'Number',
                             'Integer': 'Number'
                         },
-                        type = model.getType();
+                        type = model.getType(),
+                        options = model.get("options");
 
                     if (conversions[type]) {
                         type = conversions[type];
                     }
 
+                    if ("warning" in options) {
```

Comment:
I generally try to avoid JavaScript's `in` keyword as it can lead to a number of bugs.

Could this be switched to (the often used style):
```javascript
if (options.warning) {
    type = 'NumberWithWarning';
}
```